#Genome Assembly and Annotation
##Assembly
Genome assembly consists of 

1. data pre-processing (quality filtering and adaptor removal)
2. error correction
3. contig assembly
4. scaffolding
5. verification of scaffolds/contigs 

There is a plethora of programs that can perform some, or most of these steps. These programs include commercial and open-source options, some are very user friendly and some are extremely difficult to use/install. Common assemblers for bacterial genomes include SPADES \cite{Bankevich_2012}, MIRA \cite{Chevreux_2004}, SGA \cite{Simpson_2010}, Velvet \cite{Zerbino_2008} CLC (CLC Bio), and A5 \cite{Tritt_2012}. Good sources for overviews of genome assemblers and the assembly process include the GAGE project \cite{Salzberg_2012}, the GAGE-B project \cite{Magoc_2013}, and the Assemblathon Project \cite{Earl_2011}.

For this workflow we recommend use of the open source A5 assembly pipeline which automates all of the steps described above with a single command (\cite{Tritt\_2012}). A5 is designed to work with raw, demultiplexed Illumina data and a recent version has been optimized for longer reads from the MiSeq (Coil et al submitted). Input reads can be paired or unpaired, and the files can be separate (forward reads in one file, reverse reads in another) or interleaved. These files should have the .fastq extension. See (http://en.wikipedia.org/wiki/FASTQ_format) for a description of the fastq format. You may need assistance from your sequencing center in locating and accessing these files. You will need one of the three following (per genome): 1) a single .fastq file that contains your single reads (if paired-end sequencing was not requested), 2) a single .fastq file that contains both forward and reverse reads, or 3) two .fastq files, one with forward reads and one with the corresponding reverse reads.

Download/Install A5
Download A5 from 
http://sourceforge.net/projects/ngopt/

Follow the (expert) instructions located 
http://sourceforge.net/projects/ngopt/files/?source=navbar

or

Follow a video made by David Coil 
http://youtu.be/ePGUIj9Qbvc


or
 
Follow these instructions:

After downloading and unzipping the program, change the name of the folder to a5\_pipeline and move it from your downloads folder to your Applications folder. Then, create a new folder which will contain the files generated by the pipeline on your Desktop. By the way, there's nothing special about having your file on the Desktop, it's just there to simplify our instructions. We will refer to this folder as "a5_output", but you should use a more informative name.

###Running A5
Open a Terminal window and navigate to a5\_output. A5 will write all of the assembly output files to the same folder from which you run the program. In this example the newly created folder is on the Desktop and named a5_ouput so the syntax for navigating to the folder in a Terminal window is 

    cd Desktop/a5_output/

Now that you are in the folder where you want your genome assembly to appear, you are ready to run the program. First, type (don't hit return yet!):

    /Applications/a5_pipeline/bin/a5_pipeline.pl
    
Then, drag and drop in the input file(s) into the same Terminal window (or type the path to them if you know it). Finally, type a name that will be used as part of all of your output files. So, your command line should look like this:

    /Applications/a5_pipeline/bin/a5_pipeline.pl SequenceFile1.fastq SequenceFile2.fastq MyGenome

Once the program finishes running you will have a complete assembly located in the a5\_output folder.

Among the numerous files generated by A5, two of particular importance are the "MyGenome.contigs.fasta" and "MyGenome.final.scaffolds.fasta" which contain the contigs and scaffolds, respectively.

In addition, A5 generates a file containing information about the quality of the assembly called "assembly_stats.csv"

To view this file use the "less" command:

    less assembly_stats.csv


For more on interpreting these numbers proceed to "Assembly Validation".

###Assembly Validation
There are three components to genome assembly validation. The first is the overall "quality" of the assembly, assessed by examining the stats provided by A5 (discussed below). The second is verification that the organism sequenced is the organism of interest, simply by checking the assembled 16S sequence with BLAST. The third is "completeness" which is difficult to measure except in cases where a close reference is available. Nevertheless, we can get an idea of how complete the genome is by looking for high;y conserved "housekeeping" genes that are found in almost every bacterial genome. To do this, we use a program called Phylosift (\cite{Darling_2014}) to assess the presence or absence of 37 housekeeping genes in the assembly to infer completeness. 

###Interpretation of A5 stats
The first two numbers shown are the number of contigs and scaffolds, respectively. Defining a "good" or "bad" assembly starts here. A finished assembly would consist of a single contig but that is extremely unlikely to result from short read data. At the other extreme, we would consider a bacterial assembly in 1000 contigs to be very fragmented. In our experience, acceptable bacterial assemblies using Ilumina PE300bp data, assembled with A5, tend to range from 10-200 contigs. It is also worth noting that unless studying genomic organization, the number of contigs is less important than the gene content recovered by the assembly which is typically >99% using A5 (Coil et al, submitted).

"Genome Size" and "Longest Scaffold" are simply represented as base pairs. While genome size can vary within taxa, this can be a second sanity check for the assembly. When expecting a 5MB genome, if the assembled genome size is 2MB, a red flag should be raised. "N50" represents the contig size at which at least 50% of the assembly is contained in contigs of that size or larger.  This metric, combined with the number of contigs is the most common measure of assembly qualityâ€¦ larger is better.  An N50 of 5,000 bp would be pretty poor... meaning that half of the entire assembly is in contigs smaller than 5,000 bp.  On the other hand an N50 of 1,000,000 bp would be great for a bacterial genome.

The number of raw reads/raw nucleotides "Raw reads"/"Raw nt" and error-corrected reads/nucleotides "EC Reads"/"Raw nt" counts are useful for seeing what percentage of the data has been discarded. A very large difference between these numbers (the "Pct" stats) would indicate either poor quality input data or significant adaptor contamination. Adaptor contamination is high when the insert size is too small **(other causes?) Also, do you want to include a little troubleshooting here?**

Finally "X\_cov" shows the average coverage across the genome. For Illumina data we recommend that this number be between ~30X and 100X.  Much less than 30X coverage and the quality of any given base in the assembly may come into question.  Conversely, too much coverage can reduce the quality of the assembly and require downsampling. **Instructions or reference for downsampling?**

###Verification of 16S Sequence
Follow the steps described in Section ??, "Making a Phylogenetic Tree" for obtaining and performing a BLAST search of the full length 16s sequence.

PhyloSift
Navigate to 
http://phylosift.wordpress.com

Download and unzip the latest version of phylosift 

In the terminal, navigate to the directory containing the unzipped phylosift 

Run
    ./phylosift search contig\_file\_name

For example:
    ./phylosift search /Users/microBEnet/Desktop/Data-Genomes/Pantoea\_Tatumella/tatumella/tatumella.final.scaffolds.fasta.contigs.fsa 

Note: The first time you run PhyloSift it has to download a marker gene database so it may take a few minutes.

From the PhyloSift directory
Move to the "PS\_temp" directory

Within this directoy, Phylosift has created a directory with the same name as the input file. Move to this new directory, then move to "blastDir".

Open the marker\_summary.txt file in the blastDir

    less marker_summary.txt

The DNGNGWU0001-00040 markers represent 37 highly conserved bacterial genes, if one is missing it won't show up as a zero, it is necessary to manually verify the list. Most of the genes should only appear once. An occasional 2 is fine, but if all/a majority of the genes appear twice or even three times you have most likely sequenced multiple bacteria together. Additionally check to make sure there is no 18S RNA (at the top of the list) to ensure your sample has not been contaminated with a eukaryotes (e.g. yeast). 

Important Note: Markers 4, 8 and 38 are no longer included in the Phylosift analysis so do not be concerned if they are not listed.

##Annotation
###Options
***Define annotation and describe how it works (just like you did for assembly)***

Note that we are not describing a genome "analysis" here. While genome annotation marks the final step in our data wrangling workflow, it is just the beginning of a thorough genome analysis. We recommend performing this step as the bare-minimum analysis required to include a very basic description of the genomic content for the genome announcement publication.

*There are a number of different pipelines available for annotation of bacterial genomes. These include Prokka (\cite{Seemann_2014}), IMG (\cite{Markowitz_2014}), RAST (\cite{Overbeek_2014}), PGAP (\cite{Angiuoli_2008}) and others. 

        add glimmer http://ccb.jhu.edu/software/glimmer/index.shtml and the JCVI annotation pipeline, http://sourceforge.net/projects/jcviprok/


Each of these pipelines has advantages and disadvantages, and each will give slightly different results. Here we recommend RAST since it is web-based, easy to use, returns results within hours, and provides a convenient toolbox for analyzing the results. However, RAST annotations are very difficult to submit to NCBI so we recommend allowing NCBI to re-annotate the genome with PGAP upon submission. Also, we recommend reporting the annotation results from the PGAP annotation in the genome announcement (for consistency.) Why do we also run a RAST annotation? Because we are impatient and we like to see results right away. We do not like having to wait for the NCBI sumbission process to be completed before we start exploring our data.

###RAST Annotation
Navigate to http://rast.nmpdr.org/ and register a new account. Once you have created an account, log in. 
Hover over the "Your Jobs" tab at the top of the page and click on "Upload New Job." In order to proceed you must specify a domain, a genus, a species, and the genetic code (usually "11".) Click "Finish the Upload."

The annotation will take some time, ranging from 2 hours to a few days, depending on server load. RAST will email you when it is complete. Once the annotation is complete, use their SEED Viewer to explore the annotation and metabolic pathways of the organism. From the RAST results, you can obtain information like ***WHAT?*** 
