The objective of the present study was to design, test, troubleshoot, and publish a comprehensive workflow for microbial genome sequencing; enabling even a lab with limited resources and bioinformatics experience to perform it.

In the fall of 2011, under a grant from the Alfred P. Sloan foundation, our lab began a undergraduate research project attempting to sequence microbial reference genomes from the built environment.  Several students went through the process of isolating, identifying, sequencing and assembling microbial genomes, followed by submission to NCBI and publication of each genome (REFS).  Through the course of this project we found that while many of the steps have become much easier in recent years (e.g. kits for library prep, relatively cheap sequencing, bioinformatics pipelines), there were still a significant number of stumbling blocks.  In addition to these hurdles, some portions of the project involve choosing between a wide variety of options (e.g. choice of assembly program) which can create a large activation energy for a lab without a bioinformatician.

In the past two decades increasing advances in DNA sequencing technology have led to the creation of a wide variety of options for DNA library preparation, sequencing and assembly.  Each option comes with its own advantages and disadvantages in terms of complexity, expense, computing power, time, and experience required. In this workflow we have chosen one path through these choices, allowing a researcher to go from a swab to a published paper. The sequencing and de novo assembly of genomes has already yielded enormous scientific insight revolutionizing a diverse collection of fields, from epidemiology to ecology.   Our hope is that this workflow will help make this revolution more accessible to all scientists, as well as present educational opportunities for undergraduate researchers and classes.
 
References to similar work here?  E.g. 454 paper?

