#Library Preparation and Sequencing 
##Library Preparation
The first choice in library preparation is whether to do the library prep yourself, or to have the library made by your sequencing provider. The economics of this decision are usually dependent on the number of samples involved. For example an Illumina TruSeq library prep kit costs around $2600 for 48 samples.  That's far cheaper than the $150 to $300 that a typical sequencing provider would charge per sample.  However, if you're only preparing a couple of samples there's no reason to buy an entire kit. The requisite time and ancillary consumables and equipment must also be taken into account. Most sequencing facilities offer library preparation services.

##Kit Options
Whether you chose to make libraries yourself, or use a provider the next major choice is of the type of kit.  The two major different choices with Illumina kits are the Nextera transposase-based kits or the TruSeq kits (with or without PCR). These kits are available from Illumina, but there are also comparable options from other vendors (e.g. New England Biolabs and Kappa Bioscience). The pros and cons of each type of kit are listed below:

+ Nextera: Pro – It allows for very low amounts of input DNA, down to 1ng in the case of the Nextera XT kit. Con – the transposase has an insertion bias and the extensive PCR required for low input samples will also impact the final assembly\cite{Aird_2011}.
+ TruSeq (our recommendation): Pro – The PCR-free protocol minimizes library bias by using mechanical instead of enzymatic DNA fragmentation, and by eliminating PCR, resulting in better assemblies. Con – requires a large amount of DNA (at least 1\(\mu\)g for PCR-free).  There is also now a TruSeq LT kit which only requires 100ng of DNA but does entail some PCR so may provide a middle option between PCR-free TruSeq and Nextera.

When growing bacteria in culture, as described in this workflow, it should almost always be possible to get enough DNA to use PCR-free TruSeq and therefore minimize library preparation biases in the genome assembly.

##Considerations in Library Preparation
Insert size: The tradeoff with insert size is between utility for assembly (larger is better) and ability of those fragments to amplify on the Illumina flowcell for sequencing (smaller is better). The optimal fragment size also depends on the length of reads used (with longer read-lengths, longer insert sizes are useful for scaffolding). The final consideration is the amount of DNA available for sequencing. While having all inserts be exactly 750 base pairs (bp) might be ideal, such a stringent size-selection could result in the recovery of only a very small amount of DNA. In our lab, with paired end 300bp (PE300bp) reads on the Illumina MiSeq, we shoot for a fragment size (including adapters) of 600-900bp. Different sequencing facilities have different opinions on this topic and it is worth having a discussion with your sequencing facility's point of contact before making any libraries.

##Multiplexing
The capacity of an Ilumina MiSeq with PE300bp reads is around 15 Gigabases (Gb) which would result in a coverage of 4300X for a typical bacterium with a 3.5Mb genome. On the HiSeq with PE125bp reads, this would be over 14,000X coverage. Currently, the recommended coverage for a bacterial genome assembly is 30-100X depending on the choice of assembler. Therefore, sequencing a single bacterial genome on a full MiSeq or HiSeq run is a significant waste of money and reagents. Furthermore, some current genome assembly algorithms do not perform well given an excess of data, and require down-sampling (i.e., throwing away data) to acheive the recommended coverage for assembly. We typically multiplex 10-20 genomes on a PE300bp MiSeq run and many more on a HiSeq run. If using a kit for library prep, multiplexing is quite straightforward since there are a number of barcoded adaptors that come with the kit. Demultiplexing can be performed by the sequencing facility.

##Collaborate
Given the overcapacity of Illumina sequencing for bacterial genomes, sequencing a single genome presents a problem (unless willing to pay the ~$2000 total cost and throw away most of the data). Sequencing facilities will typically not "pool" samples from multiple groups because they don't want to oversee the pooling or deal with the associated accounting hassles. However, collaborating with other groups can be a great option.  Many labs sequence genomes or metageomes on a regular basis; adding in one additional sample isn't technically very difficult, but it will entail oversight of the pooling and the associated accounting hassles.  This will also entail a discussion of barcode compatibility, to ensure that all barcodes are sufficiently unique for demultiplexing.

##Downsampling
Coverage (read depth) is the average number of reads representing a given nucleotide and is a function of the number and size of genomes pooled onto a run.  The optimal amount of coverage depends on the read length, the assembler being used, and other factors.  For Illumina data assembled using this workflow we recommend that this number be between 20x and 200x.  See our more detailed discussion in section ??? "Interpretation of A5-miseq stats".  If you have coverage significantly higher than 200x and wish to downsample your data we have written a script (sub_sample_reads) (sub\_sample\_reads)  for this purpose. You will first need to calculate how many reads you want the script to sample. We recommend determining how many reads would be equivalent to 100x coverage (divide the genome size by the average read length and multiply by 100). You can download the script using the curl command. Create a new directory containing the reads you wish to downsample. In the terminal navigate the directory you just created and download the script using the following syntax 

    curl https://raw.githubusercontent.com/gjospin/scripts/master/subsample_reads.pl > sub_sample_reads.pl   
