## Other things to know about trees. 



A key part of interpreting phylogenetic trees lies in understanding the groupings of taxa in a tree.  Some terms relating to this are below:

***Clade*** A clade is a group of organism that share a common ancestor to the . Every node in the tree defines a unique clade.

***Monophyletic clade*** If every representative of a species is found in one single clade, and no representatives of that species are found outside of that clade, then that is a monophyletic clade. The concept of monophyly is important because all species should represent monophyletic groupings. If all species are not found in a monophyletic group, then taxonomic revision of that species is in order. Unfortunately for bacterial species, taxonomic revision is frequently in order.

***Rooting/Outgroup Selection*** The root is at the base of the tree, and is determined by the designation of an outgroup. The outgroup is an organism that you are confident is most distantly related to everything else in your tree. Outgroup selection is not a trivial task, but to simplify it here, we suggest always using an Archaeal 16S rDNA sequence as your outgroup.


***Bootstrap support*** This is a number, usually expressed on a scale of 0-100, but occasionally on a scale of 0-1, that provides an estimate of the confidence that a particular node defines a "true" clade, i.e., this specific group of organisms actually shares a common ancestor.
